common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -2418,22 +2418,26 @@ def copy_all_asset_metadata(self, source_course_key, dest_course_key, user_id):
                 # update the index entry if appropriate
                 self._update_head(dest_course_key, index_entry, dest_course_key.branch, new_structure['_id'])
 
-    def internal_clean_children(self, course_locator):
+    def fix_not_found(self, course_locator, user_id):
         """
         Only intended for rather low level methods to use. Goes through the children attrs of
-        each block removing any whose block_id is not a member of the course. Does not generate
-        a new version of the course but overwrites the existing one.
+        each block removing any whose block_id is not a member of the course.
 
         :param course_locator: the course to clean
         """
         original_structure = self._lookup_course(course_locator).structure
-        for block in original_structure['blocks'].itervalues():
+        index_entry = self._get_index_if_valid(course_locator)
+        new_structure = self.version_structure(course_locator, original_structure, user_id)
+        for block in new_structure['blocks'].itervalues():
```

Comment:
I've seen this original_structure / index_entry / new_structure pattern elsewhere. What does it do? And could it be combined into a helper method with an explanatory name?